broken windows modules path handling because of 'os.relativePath' breaking changes

'os.relativePath' got additional code:
```Nim
  when doslikeFileSystem:
    if isAbsolute(path) and isAbsolute(base):
      if not sameRoot(path, base):
        return path
```

this changes also change it's behavior:
```Nim
import os

let a = r"C:\Users\Jangko\.nimble\pkgs\eth-1.0.0\eth\common\utils.nim"
let b = r"F:\projects\bug"

debugEcho relativePath(a, b, '/')
```
* Nim 1.0.2 produce: "../../../C:/Users/Jangko/.nimble/pkgs/eth-1.0.0/eth/common/utils.nim"
* Nim 1.1.1 produce: "C:\Users\Jangko\.nimble\pkgs\eth-1.0.0\eth\common\utils.nim"

Nim compiler directly affected by this braking changes:
* in `pathutils.nim`

```Nim
  proc relativeTo*(fullPath: AbsoluteFile, baseFilename: AbsoluteDir;
                   sep = DirSep): RelativeFile =
    RelativeFile(relativePath(fullPath.string, baseFilename.string, sep))
```

* in `packagehandling.nim`
```Nim
proc fakePackageName*(conf: ConfigRef; path: AbsoluteFile): string =
  # foo-#head/../bar becomes @foo-@hhead@s..@sbar
  result = "@m" & relativeTo(path, conf.projectPath, '/').string.multiReplace({"/": "@s", "#": "@h", "@": "@@", ":": "@c"})
```

this bug actually detected before but with inconclusive ending.
*  #12420
* status-im/nimbus#424

how to reproduce:

* prepare a working windows machine + mingw gcc.
* compile nim compiler binary from `csources`.
* bootstrap a new compiler from that binary.
* __do not__ use previous version compiler with 'good' `os.relativePath` to boostrap a new compiler.

```text
$> nimble install eth
$> nimble install nimcrypto
```
then compile this file with `nim c --threads:on -d:metrics bug`

```Nim
# bug.nim
import eth/common/utils as autils, nimcrypto/utils

var x: array[100, byte]
burnMem(x)
```

### Current Output
```
C:\Users\Jangko\nimcache\bug_d\utils.nim.c.o:utils.nim.c:(.text+0x41e): multiple definition of `eth_utilsInit000'
C:\Users\Jangko\nimcache\bug_d\utils.nim.c.o:utils.nim.c:(.text+0x41e): first defined here
C:\Users\Jangko\nimcache\bug_d\@mbug.nim.c.o:@mbug.nim.c:(.text+0x225): undefined reference to `burnMem__haAOjkJjOuRy8ux9bV57dXw'
collect2.exe: error: ld returned 1 exit status
```

### Possible Solution

* `os.relativePath` need to be fixed
* or compiler internal avoid using the new 'broken' feature.

### Additional Information

* It was working in version 1.0.2
* This issue is valid only on Windows.

```
$ nim -v
Nim Compiler Version 1.1.1 [Windows: amd64]
Compiled at 2019-11-25
Copyright (c) 2006-2019 by Andreas Rumpf

git hash: 015bec625461826c3891110182b982c5dbb3208a
active boot switches: -d:release
```

--- a/compiler/packagehandling.nim
+++ b/compiler/packagehandling.nim
@@ -39,8 +39,11 @@ proc getPackageName*(conf: ConfigRef; path: string): string =
     if parents <= 0: break
 
 proc fakePackageName*(conf: ConfigRef; path: AbsoluteFile): string =
+  # Convert `path` so that 2 modules with same name
+  # in different directory get different name and they can be
+  # placed in a directory.
   # foo-#head/../bar becomes @foo-@hhead@s..@sbar
-  result = "@m" & relativeTo(path, conf.projectPath, '/').string.multiReplace({"/": "@s", "#": "@h", "@": "@@", ":": "@c"})
+  result = "@m" & relativeTo(path, conf.projectPath).string.multiReplace({$os.DirSep: "@s", $os.AltSep: "@s", "#": "@h", "@": "@@", ":": "@c"})
 
 proc demanglePackageName*(path: string): string =
   result = path.multiReplace({"@@": "@", "@h": "#", "@s": "/", "@m": "", "@c": ":"})

